uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/common/extensions/SdkExtensions.kt

```diff
@@ -173,4 +185,101 @@ fun Application.computePrimaryRoute(routePlan: RoutePlan, routeComputeCallback:
             override fun onPrimaryComputeFinished(router: Router, route: RouteInfo) = routeComputeCallback.invoke(route)
         })
     }
+}
+
+fun List<NaviSignInfo.SignElement>.concat(): String = StringBuilder().apply {
+    this@concat.forEach {
+        if (isNotEmpty()) append(", ")
+        append(it.text)
+    }
+}.toString()
+
+fun List<NaviSignInfo>.getNaviSignInfoOnRoute(): NaviSignInfo? {
+    this.filter { it.isOnRoute }.let { isOnRouteList ->
+        if (isOnRouteList.isEmpty()) {
+            return null
+        }
+
+        return isOnRouteList.firstOrNull { it.backgroundColor != 0 }?.let { it } ?: isOnRouteList[0]
```

Comment:
`NaviSignInfo` uz ma priority (neviem ci uz v release, ale ak nie, chcelo by to aspon todo)

Reply:
uff, po 4 letech vývoje se nebude muset checkovat backgroundColor? nevěřím :-D zatím tu ale vidím v SDK 14.0.0 priority jen v SignElement-u, přidám TODO.